AIRFLOW-156 Add date option to trigger_dag

lothropgauss opened the pull request:
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-156

Currently the trigger_dag command always sets the execution date to
datetime.now(). This seems like a rather arbitrary restriction and there
are use cases when running dags ad-hoc where one may wish to set a
different execution date.

Commit:
Add date option to trigger_dag

Currently the trigger_dag command always sets the execution date to
datetime.now(). This seems like a rather arbitrary restriction and there
are use cases when running dags ad-hoc where one may wish to set a
different execution date.

criccomini commented on airflow/bin/cli.py:
```diff
@@ -707,6 +707,9 @@ class CLIFactory(object):
         'conf': Arg(
             ('-c', '--conf'),
             "json string that gets pickled into the DagRun's conf attribute"),
+        'date': Arg(
```
Rather than adding this to the args for trigger_dag, can you just use the chared arg `execution_date`, above? (line 625)

lothropgauss replied:
The execution_date arg creates a required parameter, i.e. airflow run dag_id task_id execution_date. Maybe a better name for the arg could be found, but I think this should be an optional parameter. This also allows it to be compatible with the current semantics.

criccomini commented on airflow/bin/cli.py:
```diff
@@ -855,7 +858,7 @@ class CLIFactory(object):
             'func': trigger_dag,
             'help': "Trigger a DAG run",
-            'args': ('dag_id', 'subdir', 'run_id', 'conf'),
+            'args': ('dag_id', 'subdir', 'run_id', 'conf', 'date'),
```
I think all you need is execution_date here

r39132 replied:
Correct, this should be named execution_date, not date

Commit:
Update TriggerDagRunOperator for custom execution_date

Adds ability to use a custom execution_date in the TriggerDagRunOperator
and a few examples of its usage.

r39132 commented on airflow/bin/cli.py:
```diff
@@ -707,6 +707,9 @@ class CLIFactory(object):
             "JSON string that gets pickled into the DagRun's conf attribute"),
+        'date': Arg(
+            ("-d", "--execution_date"), "Override execution_date YYYY-MM-DD",
+            type=parsedate),
```
These 3 lines are not needed. Just use execution_date.
